File: src/swingjs/JSScreenDevice.java
package swingjs;

import jsjava.awt.GraphicsConfiguration;
import jsjava.awt.GraphicsDevice;


public class JSScreenDevice extends GraphicsDevice {

	/*
	 * NOTE: This class is called from jsjava.awt.GraphicsEnvironment
	 * within in j2sNative block.
	 * 
	 */
	public JSScreenDevice(){
		System.out.println("JSScreenDevice initialized");		
	}

	@Override
	public int getType() {
		return TYPE_RASTER_SCREEN;
	}

	@Override
	public String getIDstring() {
		return "swingjs.JSScreenDevice";
	}

	@Override
	public GraphicsConfiguration[] getConfigurations() {
		return new GraphicsConfiguration[] {JSToolkit.getGraphicsConfiguration()};
	}

	@Override
	public GraphicsConfiguration getDefaultConfiguration() {
		return JSToolkit.getGraphicsConfiguration();
	}
}

File: site/swingjs/j2s/swingjs/JSScreenDevice.js
Clazz.declarePackage ("swingjs");
Clazz.load (["java.awt.GraphicsDevice"], "swingjs.JSScreenDevice", ["swingjs.JSToolkit"], function () {
c$ = Clazz.declareType (swingjs, "JSScreenDevice", java.awt.GraphicsDevice);
Clazz.makeConstructor (c$, 
function () {
Clazz.superConstructor (this, swingjs.JSScreenDevice, []);
System.out.println ("JSScreenDevice initialized");
});
Clazz.overrideMethod (c$, "getType", 
function () {
return 0;
});
Clazz.overrideMethod (c$, "getIDstring", 
function () {
return "swingjs.JSScreenDevice";
});
Clazz.overrideMethod (c$, "getConfigurations", 
function () {
return [swingjs.JSToolkit.getGraphicsConfiguration ()];
});
Clazz.overrideMethod (c$, "getDefaultConfiguration", 
function () {
return swingjs.JSToolkit.getGraphicsConfiguration ();
});
});
